pkg/controllers/static/deprovisioning/controller.go

```diff
@@ -174,11 +175,75 @@ func HasNodePoolReplicaCountChanged(oldNP, newNP *v1.NodePool) bool {
 	return lo.FromPtr(oldNP.Spec.Replicas) != lo.FromPtr(newNP.Spec.Replicas)
 }
 
-// Returns nodes suitable for deprovisioning, prioritizing:
-// 1. Empty nodes (nodes with no pods or only DaemonSet pods without do-not-disrupt annotation)
-// 2. If more nodes needed, nodes with lowest disruption cost (nodes with pods that have do-not-disrupt will have highest cost)
-func (c *Controller) getDeprovisioningCandidates(ctx context.Context, np *v1.NodePool, nodes []*state.StateNode, count int) []*state.StateNode {
-	// First get empty nodes
+// Returns NodeClaims suitable for deprovisioning, prioritizing:
+// 1. Unresolved NodeClaims (no ProviderID yet - haven't launched)
+// 2. Empty nodes (nodes with no pods or only DaemonSet pods without do-not-disrupt annotation)
+// 3. If more nodes needed, nodes with lowest disruption cost (nodes with pods that have do-not-disrupt will have highest cost)
+func (c *Controller) getDeprovisioningCandidates(ctx context.Context, np *v1.NodePool, count int) []*v1.NodeClaim {
+	candidates := make([]*v1.NodeClaim, 0, count)
+
+	// Unresolved NodeClaims (haven't launched yet or that failed Create call)
+	unresolvedCandidates := c.unResolvedDeprovisioningCandidates(ctx, np.Name, count)
+	candidates = append(candidates, unresolvedCandidates...)
+	remaining := count - len(candidates)
+
+	if remaining == 0 {
+		return candidates
+	}
+
+	// Get all StateNodes for this NodePool
+	nodes := make([]*state.StateNode, 0)
+	for n := range c.cluster.Nodes() {
+		if n.Labels()[v1.NodePoolLabelKey] == np.Name && n.NodeClaim != nil && !n.MarkedForDeletion() {
+			nodes = append(nodes, n.DeepCopy())
+		}
+	}
+
+	// Resolved nodes (empty first, then by disruption cost)
+	resolvedCandidates := c.resolvedDeprovisioningCandidates(ctx, nodes, np, remaining)
+	candidates = append(candidates, resolvedCandidates...)
+
+	return candidates
+}
+
+// unResolvedDeprovisioningCandidates returns unresolved NodeClaims (those without ProviderID) up to the specified count
+func (c *Controller) unResolvedDeprovisioningCandidates(ctx context.Context, nodePoolName string, count int) []*v1.NodeClaim {
```

Comment:
nit: unresolved is one word
```suggestion
func (c *Controller) unresolvedDeprovisioningCandidates(ctx context.Context, nodePoolName string, count int) []*v1.NodeClaim {
```